src/Adapter/Product/Repository/StockAvailableRepository.php

```diff
@@ -42,6 +44,36 @@
  */
 class StockAvailableRepository extends AbstractObjectModelRepository
 {
+    /**
+     * @var CombinationRepository
+     */
+    private $combinationRepository;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @param CombinationRepository $combinationRepository
+     * @param Connection $connection
+     * @param string $dbPrefix
+     */
+    public function __construct(
+        CombinationRepository $combinationRepository,
+        Connection $connection,
+        string $dbPrefix
+    ) {
+        $this->combinationRepository = $combinationRepository;
```

Comment:
I believe you don't use this one any more because now you rely on a custom query built based only on combination ID

I wonder if we should fetch the combination's product first (this would at least assure its existence), and rely on the `StockAvailable::getStockAvailableIdByProductId`
The advantage is that it handles shop restrictions internally, but can we rely on this? ^^

Reply:
Is it necessary?
combination should not exist if there is no product and stock should not exist if there is no combination, or its not how it works?
Combination is a separate Entity, it seems unnecessary|overhead to check its product existence each time we deal with it

Reply:
Yeah I guess that makes sense, and the repo has been cleared so we're good